pkg/rime/rime.go

```diff
@@ -0,0 +1,133 @@
+package rime
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"time"
+)
+
+const (
+	EmptyData  uint16 = 0
+	StringData uint16 = 1
+
+	StatusOK  uint16 = 200
+	StatusErr uint16 = 500
+)
+
+type Payload struct {
+	Type      uint16
+	Length    uint16
+	Timestamp uint32
+	Data      []byte
+}
+
+type Writer struct {
+	writer io.Writer
+}
+
+type Reader struct {
+	reader io.Reader
+}
+
+func NewWriter(w io.Writer) *Writer {
+	return &Writer{writer: w}
+}
+
+func NewReader(r io.Reader) *Reader {
+	return &Reader{reader: r}
+}
+
+func (p *Payload) ToString() string {
+	return fmt.Sprintf(
+		"[TYP: %v | LEN: %v | TIM: %v | DAT: %v]",
+		p.Type,
+		p.Length,
+		time.Unix(int64(p.Timestamp), 0),
+		string(p.Data),
+	)
+}
+
+func (w *Writer) Write(p *Payload) error {
+	err := binary.Write(w.writer, binary.LittleEndian, p.Type)
+	if err != nil {
+		return err
+	}

```

Comment:
Here's another way to write it. Choose freely which you like more.

```suggestion
	if err := binary.Write(w.writer, binary.LittleEndian, p.Type); err != nil {
		return err
	}

```